### ISSUE TITLE
12:1: Error: Null safe libraries are not allowed to export declarations from of opt-out libraries.

### ISSUE THREAD

**Issue by imeDevelopers** [flutter/flutter#61596]
```
[+9046 ms] [+18979 ms] ../../flutter/packages/flutter/lib/foundation.dart:12:1: Error: Null
safe libraries are not allowed to export declarations from of opt-out libraries.
[  +20 ms] [  +16 ms] export 'package:meta/meta.dart' show
[   +4 ms] [   +2 ms] ^
[+78475 ms] [+78482 ms] Persisting file store
[ +112 ms] [  +61 ms] Done persisting file store
[+1129 ms] [+1054 ms] Target kernel_snapshot failed: Exception
[  +77 ms]            #0      KernelSnapshot.build
(package:flutter_tools/src/build_system/targets/common.dart:262:7)
[+11314 ms]            <asynchronous suspension>
[ +248 ms]            #1      _BuildInstance._invokeInternal
(package:flutter_tools/src/build_system/build_system.dart:795:27)
[ +208 ms]            #2      _rootRunUnary (dart:async/zone.dart:1198:47)
[  +12 ms]            #3      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[   +6 ms]            #4      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[   +8 ms]            #5      Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[ +226 ms]            #6      Future._propagateToListeners (dart:async/future_impl.dart:725:32)[  +19 

ms]            #7      Future._completeWithValue (dart:async/future_impl.dart:529:5)
[   +5 ms]            #8      Future._asyncCompleteWithValue.<anonymous closure>
(dart:async/future_impl.dart:567:7)
[  +85 ms]            #9      _rootRun (dart:async/zone.dart:1190:13)
[   +6 ms]            #10     _CustomZone.run (dart:async/zone.dart:1093:19)
[  +61 ms]            #11     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
[   +4 ms]            #12     _CustomZone.bindCallbackGuarded.<anonymous closure>
(dart:async/zone.dart:1037:23)
[  +96 ms]            #13     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)        
[  +14 ms]            #14     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[ +260 ms]            #15     _runPendingImmediateCallback
(dart:isolate-patch/isolate_patch.dart:118:13)
[ +114 ms]            #16     _RawReceivePortImpl._handleMessage
(dart:isolate-patch/isolate_patch.dart:169:5)
[ +682 ms] [  +56 ms] 
[  +93 ms]            #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)  
[  +12 ms]            #1      AssembleCommand.runCommand
(package:flutter_tools/src/commands/assemble.dart:238:7)
[ +280 ms]            #2      _rootRunUnary (dart:async/zone.dart:1198:47)
[ +118 ms]            #3      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[  +11 ms]            #4      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[  +80 ms]            #5      Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[  +97 ms]            #6      Future._propagateToListeners (dart:async/future_impl.dart:725:32)[  +73 

ms]            #7      Future._completeWithValue (dart:async/future_impl.dart:529:5)    
[  +10 ms]            #8      _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
[   +9 ms]            #9      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)[   +7 

ms]            #10     FlutterBuildSystem.build
(package:flutter_tools/src/build_system/build_system.dart)
[   +9 ms]            #11     _rootRunUnary (dart:async/zone.dart:1198:47)
[   +5 ms]            #12     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[   +5 ms]            #13     _FutureListener.handleValue (dart:async/future_impl.dart:143:18) 
[   +8 ms]            #14     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[  +11 ms]            #15     Future._propagateToListeners (dart:async/future_impl.dart:725:32)[   +7 

ms]            #16     Future._completeWithValue (dart:async/future_impl.dart:529:5)    
[   +5 ms]            #17     _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
[  +54 ms]            #18     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)[   +6 

ms]            #19     _BuildInstance.invokeTarget
(package:flutter_tools/src/build_system/build_system.dart)
[   +8 ms]            #20     _rootRunUnary (dart:async/zone.dart:1198:47)
[   +4 ms]            #21     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[  +68 ms]            #22     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[  +59 ms]            #23     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[ +119 ms]            #24     Future._propagateToListeners (dart:async/future_impl.dart:725:32)[  +93 

ms]            #25     Future._completeWithValue (dart:async/future_impl.dart:529:5)
[ +124 ms]            #26     Future.wait.<anonymous closure> (dart:async/future.dart:410:22)
[  +99 ms]            #27     _rootRunUnary (dart:async/zone.dart:1198:47)
[ +156 ms] [ +195 ms] "flutter assemble" took ١٠٠٬٧٨٣ms.
[ +470 ms]            #28     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[  +33 ms]            #29     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[   +5 ms]            #30     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[  +28 ms]            #31     Future._propagateToListeners (dart:async/future_impl.dart:725:32)[   +5 

ms]            #32     Future._completeWithValue (dart:async/future_impl.dart:529:5)
[  +17 ms]            #33     _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
[  +10 ms]            #34     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)[   +6 

ms]            #35     _BuildInstance._invokeInternal
(package:flutter_tools/src/build_system/build_system.dart)
[   +8 ms]            #36     _asyncErrorWrapperHelper.errorCallback
(dart:async-patch/async_patch.dart:91:64)
[  +96 ms]            #37     _rootRunBinary (dart:async/zone.dart:1214:47)
[  +55 ms]            #38     _CustomZone.runBinary (dart:async/zone.dart:1107:19)
[   +5 ms]            #39     _FutureListener.handleError (dart:async/future_impl.dart:157:20)
[ +156 ms]            #40     Future._propagateToListeners.handleError
(dart:async/future_impl.dart:708:47)
[ +126 ms]            #41     Future._propagateToListeners (dart:async/future_impl.dart:729:24)[  +82 

ms]            #42     Future._completeError (dart:async/future_impl.dart:537:5)
[   +5 ms]            #43     _AsyncAwaitCompleter.completeError
(dart:async-patch/async_patch.dart:47:15)
[  +62 ms]            #44     KernelSnapshot.build
(package:flutter_tools/src/build_system/targets/common.dart)
[  +69 ms]            #45     _rootRunUnary (dart:async/zone.dart:1198:47)
[  +55 ms]            #46     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[  +90 ms]            #47     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[   +5 ms]            #48     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[ +206 ms]            #49     Future._propagateToListeners (dart:async/future_impl.dart:725:32)[   +7 

ms]            #50     Future._completeWithValue (dart:async/future_impl.dart:529:5)    
[   +7 ms]            #51     _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
[  +81 ms]            #52     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)[   +7 

ms]            #53     KernelCompiler.compile (package:flutter_tools/src/compile.dart)
[  +61 ms]            #54     _rootRunUnary (dart:async/zone.dart:1198:47)
[   +7 ms]            #55     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
[  +82 ms]            #56     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
[   +5 ms]            #57     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
[ +162 ms]            #58     Future._propagateToListeners (dart:async/future_impl.dart:725:32)[ +105 

ms]            #59     Future._completeWithValue (dart:async/future_impl.dart:529:5)    
[ +181 ms]            #60     Future._asyncCompleteWithValue.<anonymous closure>
(dart:async/future_impl.dart:567:7)
[ +122 ms]            #61     _rootRun (dart:async/zone.dart:1190:13)
[  +96 ms]            #62     _CustomZone.run (dart:async/zone.dart:1093:19)
[   +6 ms]            #63     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
[   +6 ms]            #64     _CustomZone.bindCallbackGuarded.<anonymous closure>
(dart:async/zone.dart:1037:23)
[  +11 ms]            #65     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
[   +9 ms]            #66     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[  +88 ms]            #67     _runPendingImmediateCallback
(dart:isolate-patch/isolate_patch.dart:118:13)
[  +15 ms]            #68     _RawReceivePortImpl._handleMessage
(dart:isolate-patch/isolate_patch.dart:169:5)
[  +11 ms] FAILURE: Build failed with an exception.
[ +129 ms] * Where:
[  +13 ms] Script 'E:\Android\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 896
[ +117 ms] * What went wrong:
[   +5 ms] Execution failed for task ':app:compileFlutterBuildDebug'.
[ +425 ms] > Process 'command 'E:\Android\flutter\bin\flutter.bat'' finished with non-zero exitvalue 1
[   +7 ms] * Try:
[   +9 ms] Run with --stacktrace option to get the stack trace. Run with --info or --debug     
option to get more log output. Run with --scan to get full insights.
[   +9 ms] * Get more help at https://help.gradle.org
[   +5 ms] BUILD FAILED in 10m 48s
[   +9 ms] [  +72 ms] ensureAnalyticsSent: 10ms
[   +5 ms] [  +16 ms] Running shutdown hooks
[   +4 ms] [   +2 ms] Shutdown hooks complete
[   +5 ms] [   +7 ms] exiting with code 1
[   +4 ms] > Task :app:compileFlutterBuildDebug FAILED
[   +3 ms] w: Detected multiple Kotlin daemon sessions at ..\build\kotlin\sessions
[   +4 ms] Deprecated Gradle features were used in this build, making it incompatible with
Gradle 7.0.
[   +6 ms] Use '--warning-mode all' to show the individual deprecation warnings.
[ +131 ms] See
https://docs.gradle.org/6.1.1/userguide/command_line_interface.html#sec:command_line_warnings  
[ +319 ms] 1012 actionable tasks: 71 executed, 941 up-to-date
[ +329 ms] Running Gradle task 'assembleDebug'... (completed in ٦٧١٫٩s, longer than expected)
[ +207 ms] Exception: Gradle task assembleDebug failed with exit code 1
[ +245 ms] "flutter run" took ٦٨٣٬١١٣ms.
[ +138 ms] 
                    #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)    
                    #1      RunCommand.runCommand
(package:flutter_tools/src/commands/run.dart:593:7)
                    <asynchronous suspension>
                    #2      FlutterCommand.verifyThenRunCommand
                    (package:flutter_tools/src/runner/flutter_command.dart:872:18)
                    #3      _rootRunUnary (dart:async/zone.dart:1198:47)
                    #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                    #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)   
                    #6      Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
                    #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)  
                    #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)      
                    #9      _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
                    #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)  
                    #11     RunCommand.usageValues
(package:flutter_tools/src/commands/run.dart)
                    #12     _rootRunUnary (dart:async/zone.dart:1198:47)
                    #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                    #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)   
                    #15     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
                    #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)  
                    #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)      
                    #18     _AsyncAwaitCompleter.complete
(dart:async-patch/async_patch.dart:40:15)
                    #19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)  
                    #20     AndroidDevice.isLocalEmulator
(package:flutter_tools/src/android/android_device.dart)
                    #21     _rootRunUnary (dart:async/zone.dart:1198:47)
                    #22     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                    #23     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)   
                    #24     Future._propagateToListeners.handleValueCallback
(dart:async/future_impl.dart:696:45)
                    #25     Future._propagateToListeners (dart:async/future_impl.dart:725:32)  
                    #26     Future._completeWithValue (dart:async/future_impl.dart:529:5)      
                    #27     Future._asyncCompleteWithValue.<anonymous closure>
(dart:async/future_impl.dart:567:7)
                    #28     _rootRun (dart:async/zone.dart:1190:13)
                    #29     _CustomZone.run (dart:async/zone.dart:1093:19)
                    #30     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
                    #31     _CustomZone.bindCallbackGuarded.<anonymous closure>
(dart:async/zone.dart:1037:23)
                    #32     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
                    #33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)      
                    #34     _runPendingImmediateCallback
(dart:isolate-patch/isolate_patch.dart:118:13)
                    #35     _RawReceivePortImpl._handleMessage
(dart:isolate-patch/isolate_patch.dart:169:5)


[+2829 ms] ensureAnalyticsSent: 268ms
[ +133 ms] Running shutdown hooks
[ +255 ms] Shutdown hook priority 4
[  +64 ms] Shutdown hooks complete
[ +189 ms] exiting with code 1
```

-------------------------------------------------
flutter doctor -v
```
[√] Flutter (Channel master, 1.20.0-8.0.pre.125, on Microsoft Windows [Version 10.0.18363.959],    

locale ar-EG)
    • Flutter version 1.20.0-8.0.pre.125 at E:\Android\flutter
    • Framework revision 9c4a5ef1ed (10 hours ago), 2020-07-15 21:41:27 +0100
    • Engine revision 774ae0313d
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 89fe12b12a)

 
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\hp\AppData\Local\Android\Sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\Users\hp\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Users\hp\AppData\Local\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.47.1)
    • VS Code at C:\Users\hp\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.11.0

[√] Connected device (3 available)
    • RMX1911 (mobile) • 9fb8e7d7   • android-arm64  • Android 9 (API 28)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 83.0.4103.116

• No issues found!
```

it's fatal errors , that i should wait you to fix it to be able to develop my app ..

**Comment by TahaTesser** [comment]
Hi @imeDevelopers 
Can you please provide your `flutter run --verbose`,  your  `pubspec.yaml` and a complete reproducible minimal code sample
Thank you

**Comment by imeDevelopers** [comment]
@TahaTesser 
the error itself tells you that this is flutter error in file `../../flutter/packages/flutter/lib/foundation.dart`, no need to know my code ..

i fixed this problem using `flutter downgrade` ...

**Comment by jonahwilliams** [comment]
@imeDevelopers can you run `flutter pub get` again to see if the issue persists? If so, please look in `.dart_tool/package_config` and report the version of `meta` that is there

**Comment by liaobushi520** [comment]
1:clean
2:flutter pub get

**Comment by darshankawar** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by visual-velocity** [comment]
How to resolved this issue in Flutter 1.22 & Dart 2.10

**Comment by justkawal** [comment]
@visual-velocity I am having the same issue.

**Comment by ericmartineau** [comment]
I had a `dependency_overrides` entry for `meta: 1.3.2` which wasn't compatible with the newer null-safety flutter code.
Removing that dependency solved this problem for me.  Flutter `1.22.1` dart `2.10`.  

FWIW, the package resolver chose `meta-1.3.0-nullsafety.3`

**Comment by celesteking** [comment]
Why was this closed? Reproduce this by installing flutter 1.20 , `flutter create test`, then upgrade flutter and see the errors on building.

**Comment by mohsinabdulsultan** [comment]
Yeah ,` flutter-downgrade` command work very well on Mac. Go through that one , It must help !
